### PR TITLE
NT-813 - Approvals history

### PR DESCRIPTION
**Pull request by reg0**
this PR is based on https://github.com/SopraSteriaAS/TimeforingFrontend/pull/319 so that one should be merged first

it's also partially dependent on other approvals related stuff, so further refactorings may be necessary to remove code duplications

in order to test it, my PR in backend has to be merged first and deployed to test env.

### CODE REVIEW

File: src/pages/approval-history/approval-history.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
+import { Component, ChangeDetectorRef, ChangeDetectionStrategy } from '@angular/core';
+import { trigger, style, animate, transition } from '@angular/animations';
+import { IonicPage, NavController, NavParams, PopoverController } from 'ionic-angular';
+import { NgRedux, select } from 'ng2-redux';
+import { Observable } from 'rxjs';
+import * as moment from 'moment';
+import { IAppState } from '../../interfaces/IAppState';
+import { EntryForApprovalByPM, ApprovalUser, ProjectForApproval, PeriodForApproval, ApprovedEntry } from '../../models'; 
+import { TimeEntryService, FormattingService, SaveAnimationService, ApprovalService } from '../../services';
+import { } from '../../services';
```
Comment (hasnik) [inline src/pages/approval-history/approval-history.ts]: Small mistake ;)

File: src/pages/approval-history/approval-history.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  }
+
+  toggleSection(obj: any) {
+    if (this.isSectionOpen(obj.projectNo, obj.resourceNo, obj.startDate)) {
+      this.openSection = {projectNo: '', resourceNo: '', periodStartDate: ''};
+      return;
+    }
+    let {projectNo, resourceNo} = obj;
+    this.openSection = {projectNo, resourceNo, periodStartDate: this.formattingService.getFormattedDateString(obj.startDate)};
+    // this.cdr.markForCheck()
```
Comment (hasnik) [inline src/pages/approval-history/approval-history.ts]: Remove comment?

File: src/pages/approval/approval-pm/approval-pm.html [hunk 3]
```diff
@@ -2,8 +2,8 @@
     <ng-container *ngIf="entriesForApproval">
       <div class="project-container" *ngFor="let project of getProjects()">
         <div class="project-header">
-            <h5 class="project-customer">{{project.customerName}}</h5>
-            <h5 class="project-description"><b>{{project.name}}</b></h5>
+          <h5 class="project-customer">{{project.customerName}}</h5>
+          <h5 class="project-description"><b>{{project.name}}</b></h5>
```
Comment (hasnik) [inline src/pages/approval/approval-pm/approval-pm.html]: I believe bold should be added with project-description style using font-weight property

Comment (reg0) [reply src/pages/approval/approval-pm/approval-pm.html]: it was there earlier, but I totally agree, thanks

**Pull request merged**